### DISCUSSION TITLE
torch_geometric.compile for MetaLayers with scatter operations

### DISCUSSION THREAD

**Discussion by mario-linov** [pyg-team/pytorch_geometric#7389] [General]
I was having a look at this tutorial [https://pytorch-geometric.readthedocs.io/en/latest/tutorial/compile.html](https://pytorch-geometric.readthedocs.io/en/latest/tutorial/compile.html) about torch_geometric.compile.

I do not understand properly the implications of this for scatter operations from the package torch_scatter. Are these operations executed without compilation or GNN models with these operations cannot be compiled at all?

Scatter operations are needed in MetaLayer [https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.models.MetaLayer.html?highlight=meta#torch_geometric.nn.models.MetaLayer](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.models.MetaLayer.html?highlight=meta#torch_geometric.nn.models.MetaLayer), which are quite common building blocks of GNNs.

**Comment by rusty1s** [comment]
Sorry for not being clear on this one. We have upstreamed `torch-scatter` into native PyTorch, and bundled them all together in `torch_geometric.utils.scatter`. You can still use `scatter` operations and see the expected speed-ups if you use `torch_geometric.utils.scatter`.
